pgo/cmd/cluster.go

```diff
@@ -504,6 +528,29 @@ func isTablespaceParam(param string) bool {
 	return found
 }
 
+// setClusterAnnotationGroup sets up the annotations for a particular group
+func setClusterAnnotationGroup(annotationGroup map[string]string, annotations []string) {
+	for _, annotation := range annotations {
+		// there are two types of annotations syntaxes:
+		//
+		// 1: key=value (adding, editing)
+		// 2: key- (removing)
+		if strings.HasSuffix(annotation, "-") {
+			annotationGroup[strings.TrimSuffix(annotation, "-")] = ""
+			continue
+		}
+
+		parts := strings.Split(annotation, "=")
+
+		if len(parts) != 2 {
```

Comment:
We could leverage some of the validation logic included in `apimachinery` to further/fully validate the annotation key/values provided: https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go

Reply:
I may be missing something, but I don't see anything that validates the Annotation format in there.

Reply:
Add most I believe it would be `IsValidLabelValue`?